docs/builtins.md

```diff
@@ -0,0 +1,730 @@
+
+#### Example
+
+```evy
+map := {a:1}
+printf "has %v %q: %t\n" map "a" (has map "a")
+printf "has %v %q: %t\n" map "X" (has map "X")
+```
+
+Output
+```
+has {a:1} "a": true
+has {a:1} "X": false
+```
+
+#### Reference 
+
+    has:bool map:{} key:string
+
+The `has` function takes two arguments: a map and a key. The function
+returns true if the map has the key, and false if the map does not have
+the key. The map can have any value type, such as `{}num` or `{}
+[]any` and the key can be any string.
+
+### `del`
+
+`del` deletes a key-value entry from a map.
+
+#### Example
+
+```evy
+map := {a:1 b:2}
+del map "b"
+print map
+```
+
+Output
+```
+{a:1}
+```
+
+#### Reference 
+
+    del map:{} key:string
+
+The `del` function takes two arguments: a map and a key. It deletes the
+key-value entry from the map if the key exists. If the key does not
+exist, the function does nothing. The map can have any value type, and
+the key can be any string.
+
+## Program control
+
+### `sleep`
+
+`sleep` pauses the program for the number of seconds given. 
+
+`sleep` can be used to create delays in Evy programs. For example, you
+could use sleep to create a countdown timer or to wait for a user to
+input a value.
+
+#### Example
+
+```evy
+print "1"
+sleep 1
+print "2"
+```
+
+Output
+```
+1
+2
+```
+
+#### Reference 
+
+    sleep seconds:num
+
+The `sleep` function pauses the execution of the current Evy program for
+at least the number of seconds given. Sleep may also pause for a
+fraction of a second, e.g. `sleep 0.1`.
+
+### `exit`
+
+`exit` terminates the program with the given status code.
+
+#### Example
+
+```evy
+print "1"
+sleep 1
+print "2"
+```
+
+Output
+```
+1
+2
+```
+
+#### Reference 
+
+    exit status:num
+
+The `exit` function takes a single argument, which is the status code
+that the program will terminate with. The status code can be any
+number, but it is typically used to indicate whether the program
+terminated successfully or with an error. A status code of 0 means that
+the program terminated successfully, while any other status code is
+considered an error.
+
+In the Evy web interface, if the exit function is called with a status
+code greater than 0, the interface will highlight the line where the
+exit function is called and print the status code to the output area.
+This makes it easy to identify the source of any errors that occur in
+your program.
+
+## Conversion
+
+### `str2num`
+
+`str2num` converts a string to a number. If the string is not a valid
+number, it returns `0` and sets the global `err` to true.
+
+#### Example
+
+```evy
+n:num
+n = str2num "1"
+print "n:" n "err:" err
+n = str2num "NOT-A-NUMBER"
+print "n:" n "err:" err
+```
+
+Output
+```
+n: 1 err: false
+n: 0 err: true
+```
+
+#### Reference 
+
+    str2num:num s:string
+
+TBD
+
+### `str2bool`
+
+`str2bool` converts a string to a boolean. If the string is not a valid
+boolean, it returns `0` and sets the global `err` to true.
```

Comment:
same here: global variable

Reply:
pre-declared, global variable ?
fixed with what you suggested.